optimizely/client/client.go

```diff
@@ -36,19 +37,28 @@ type OptimizelyClient struct {
 }
 
 // IsFeatureEnabled returns true if the feature is enabled for the given user
-func (o *OptimizelyClient) IsFeatureEnabled(featureKey string, userContext entities.UserContext) (bool, error) {
+func (o *OptimizelyClient) IsFeatureEnabled(featureKey string, userContext entities.UserContext) (result bool, err error) {
 	if !o.isValid {
 		errorMessage := "Optimizely instance is not valid. Failing IsFeatureEnabled."
 		err := errors.New(errorMessage)
 		logger.Error(errorMessage, nil)
 		return false, err
 	}
 
+	defer func() {
+		if r := recover(); r != nil {
+			errorMessage := fmt.Sprintf(`Optimizely SDK is panicking with the error "%s"`, string(debug.Stack()))
+			err = errors.New(errorMessage)
+			logger.Error(errorMessage, err)
+			result = false
```

Comment:
why are you assigning result to false? what is the purpose of it?

Reply:
You are right, this is actually not needed because the zero-value is `false`